lib/table.h

```diff
@@ -275,12 +275,16 @@ static inline struct route_node *route_table_iter_next(route_table_iter_t *iter)
 		break;
 
 	case RT_ITER_STATE_PAUSED:
+	{
+		union prefixconstptr cp = {.p = &iter->pause_prefix};
```

Comment:
This is _extremely_ ugly... the whole point of `union prefixconstptr` is that it is tagged with `__attribute__((transparent_union))` which makes gcc accept any of the union member types in its place when used as a function parameter.  The idea is that this type is used in headers (and only there) on all functions that should accept either `struct prefix`, `struct prefix_ipv4`, `struct prefix_ipv6` or any other of the prefix family.

Are you sure this can't be made to work with C++ somehow?  The gcc docs don't say anything about this being a C-only attribute (but maybe it just doesn't work in C++?)

Reply:
yeah, don't love it myself... I'll poke around, try some things, and see if I can make this one more palatable.

Reply:
So from what I'm reading, that gcc extension just doesn't apply to c++. I think this ... is functional (but probably I should add a comment explaining the gymnastics). Other than moving this inline back into table.c, so it's not seen at compile time, I'm not sure what would be ... 'better'.

Reply:
Yes, g++ doesn't support `__attribute__((transparent_union))` so I think there isn't much we can do about this (would love to be proved wrong!).

One thing that works is adding a static cast like this:
```c
                node = route_table_get_next(
                        iter->table,
                        *(union prefixconstptr *)&iter->pause_prefix);
```

But of course this is also ugly.

Maybe the best solution is to uninline `route_table_get_next()` and move it to table.c as Mark suggested. The performance benefit of inline functions should be negligible on modern hardware, and they make it harder to make our headers compatible with C++. Thoughts?